The Rosbag2 tests using the recorder tend to be flaky on CI due to the small recorder cache size

## Description
The Rosbag2 tests using the recorder tend to be flaky on CI due to the small recorder cache size.

### Expected Behavior
Tests shall not fail with the occasional messages lost due to the slow storage on CI.

### Actual Behavior
The recorder tests some time fails due to the messages lost in recorder because of the cache overflow.

## To Reproduce
<details><summary>Example of the test failure. Click to expand</summary>

```
[ RUN      ] TestRecordEndToEnd/RecordFixture.record_fails_if_neither_all_nor_topic_list_are_specified/1
[ERROR] [ros2bag]: Need to specify at least one option out of --all, --all-topics, --all-services, --services, --topics, --topic-types,--static-topics-path or --regex
...
...
Total lost: 2
[INFO] [1752779528.183694324] [rosbag2_recorder]: Event publisher thread: Exiting
[INFO] [1752779528.184001492] [rosbag2_recorder]: Recording stopped
[INFO] [1752779528.184223648] [rclcpp]: signal_handler(signal_value=2)
[INFO] [1752779528.498942582] [rosbag2_storage]: Opened database '/bb/exchange/worker-build/d65a0ad07d4f2958/root/_tmp/6203efcd52a01311d4488813eb683adf/tmp_test_dir_fbf7ca/record_end_to_end_test_with_cache_0_sqlite3/record_end_to_end_test_with_cache_0_sqlite3_0.db3' for READ_ONLY.
grace/recording/rosbag2/rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp:705: Failure
Value of: test_topic_messages
Expected: size is >= 33
  Actual: { 16-byte object <F0-8F D2-7C B5-55 00-00 E0-8F D2-7C B5-55 00-00>, 16-byte object <C0-1A E9-7C B5-55 00-00 B0-1A E9-7C B5-55 00-00>, 16-byte object <30-AD FF-7C B5-55 00-00 20-AD FF-7C B5-55 00-00>, 16-byte object <00-45 EC-7C B5-55 00-00 F0-44 EC-7C B5-55 00-00>, 16-byte object <40-37 FD-7C B5-55 00-00 30-37 FD-7C B5-55 00-00>, 16-byte object <E0-FC FF-7C B5-55 00-00 D0-FC FF-7C B5-55 00-00>, 16-byte object <80-4F E9-7C B5-55 00-00 70-4F E9-7C B5-55 00-00>, 16-byte object <70-F1 25-7D B5-55 00-00 60-F1 25-7D B5-55 00-00>, 16-byte object <E0-ED 25-7D B5-55 00-00 D0-ED 25-7D B5-55 00-00>, 16-byte object <00-EC 25-7D B5-55 00-00 F0-EB 25-7D B5-55 00-00>, 16-byte object <C0-4C FD-7C B5-55 00-00 B0-4C FD-7C B5-55 00-00>, 16-byte object <D0-F8 FF-7C B5-55 00-00 C0-F8 FF-7C B5-55 00-00>, 16-byte object <80-59 EF-7C B5-55 00-00 70-59 EF-7C B5-55 00-00>, 16-byte object <30-6C EE-7C B5-55 00-00 20-6C EE-7C B5-55 00-00>, 16-byte object <F0-E6 25-7D B5-55 00-00 E0-E6 25-7D B5-55 00-00>, 16-byte object <70-BA EE-7C B5-55 00-00 60-BA EE-7C B5-55 00-00>, 16-byte object <90-B0 EE-7C B5-55 00-00 80-B0 EE-7C B5-55 00-00>, 16-byte object <30-57 FD-7C B5-55 00-00 20-57 FD-7C B5-55 00-00>, 16-byte object <30-59 FD-7C B5-55 00-00 20-59 FD-7C B5-55 00-00>, 16-byte object <40-E5 FF-7C B5-55 00-00 30-E5 FF-7C B5-55 00-00>, 16-byte object <40-E7 FF-7C B5-55 00-00 30-E7 FF-7C B5-55 00-00>, 16-byte object <40-E9 FF-7C B5-55 00-00 30-E9 FF-7C B5-55 00-00>, 16-byte object <F0-CF F8-7C B5-55 00-00 E0-CF F8-7C B5-55 00-00>, 16-byte object <F0-D1 F8-7C B5-55 00-00 E0-D1 F8-7C B5-55 00-00>, 16-byte object <F0-D3 F8-7C B5-55 00-00 E0-D3 F8-7C B5-55 00-00>, 16-byte object <F0-15 F9-7C B5-55 00-00 E0-15 F9-7C B5-55 00-00>, 16-byte object <F0-17 F9-7C B5-55 00-00 E0-17 F9-7C B5-55 00-00>, 16-byte object <F0-19 F9-7C B5-55 00-00 E0-19 F9-7C B5-55 00-00>, 16-byte object <E0-12 E9-7C B5-55 00-00 D0-12 E9-7C B5-55 00-00>, 16-byte object <C0-14 E9-7C B5-55 00-00 B0-14 E9-7C B5-55 00-00>, 16-byte object <C0-16 E9-7C B5-55 00-00 B0-16 E9-7C B5-55 00-00> }, whose size 31 doesn't match
[  FAILED  ] TestRecordEndToEnd/RecordFixture.record_end_to_end_test_with_cache/0, where GetParam() = "sqlite3" (5570 ms)
```
</details>


## System (please complete the following information)
 - OS: Ubuntu 22.04
 - ROS 2 Distro: Rolling
 - Install Method: Build from source on CI
 - Version: 85d07432cb6f4f34800d007e13c9596209268c95


## Additional context
N/A

--- a/rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp
+++ b/rosbag2_compression/test/rosbag2_compression/test_sequential_compression_writer.cpp
@@ -631,7 +631,7 @@ TEST_P(SequentialCompressionWriterTest, split_event_calls_callback_with_file_com
 TEST_F(SequentialCompressionWriterTest, snapshot_writes_to_new_file_with_file_compression)
 {
   tmp_dir_storage_options_.max_bagfile_size = 0;
-  tmp_dir_storage_options_.max_cache_size = 200;
+  tmp_dir_storage_options_.max_cache_size = 700;
   tmp_dir_storage_options_.snapshot_mode = true;
 
   initializeFakeFileStorage();

--- a/rosbag2_cpp/test/rosbag2_cpp/test_sequential_writer.cpp
+++ b/rosbag2_cpp/test/rosbag2_cpp/test_sequential_writer.cpp
@@ -520,7 +520,7 @@ TEST_F(SequentialWriterTest, do_not_use_cache_if_cache_size_is_zero) {
 TEST_F(SequentialWriterTest, snapshot_mode_write_on_trigger)
 {
   storage_options_.max_bagfile_size = 0;
-  storage_options_.max_cache_size = 200;
+  storage_options_.max_cache_size = 700;
   storage_options_.snapshot_mode = true;
 
   // Expect a single write call when the snapshot is triggered
@@ -552,7 +552,7 @@ TEST_F(SequentialWriterTest, snapshot_mode_write_on_trigger)
 TEST_F(SequentialWriterTest, snapshot_mode_not_triggered_no_storage_write)
 {
   storage_options_.max_bagfile_size = 0;
-  storage_options_.max_cache_size = 200;
+  storage_options_.max_cache_size = 700;
   storage_options_.snapshot_mode = true;
 
   // Storage should never be written to when snapshot mode is enabled
@@ -598,10 +598,10 @@ TEST_F(SequentialWriterTest, snapshot_mode_zero_cache_size_throws_exception)
 TEST_F(SequentialWriterTest, snapshot_writes_to_new_file_with_bag_split)
 {
   storage_options_.max_bagfile_size = 0;
-  storage_options_.max_cache_size = 200;
+  storage_options_.max_cache_size = 700;
   storage_options_.snapshot_mode = true;
   const rcutils_time_point_value_t first_msg_timestamp = 100;
-  const size_t num_msgs_to_write = 100;
+  const size_t num_msgs_to_write = 150;
   const std::string topic_name = "test_topic";
   std::string msg_content = "Hello";
   const size_t serialized_msg_buffer_length = msg_content.length();
@@ -700,7 +700,7 @@ TEST_F(SequentialWriterTest, snapshot_writes_to_new_file_with_bag_split)
 TEST_F(SequentialWriterTest, snapshot_can_be_called_twice)
 {
   storage_options_.max_bagfile_size = 0;
-  storage_options_.max_cache_size = 200;
+  storage_options_.max_cache_size = 700;
   storage_options_.snapshot_mode = true;
   const size_t num_msgs_to_write = 100;
 

--- a/rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp
+++ b/rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp
@@ -582,7 +582,7 @@ TEST_P(RecordFixture, record_fails_gracefully_if_plugin_for_given_encoding_does_
 }
 
 TEST_P(RecordFixture, record_end_to_end_test_with_cache) {
-  auto max_cache_size = 10;
+  auto max_cache_size = 700;
   auto topic_name = "/rosbag2_cache_test_topic";
 
   auto message = get_messages_strings()[0];

--- a/rosbag2_transport/test/rosbag2_transport/test_record_services.cpp
+++ b/rosbag2_transport/test/rosbag2_transport/test_record_services.cpp
@@ -71,7 +71,7 @@ public:
     rosbag2_transport::RecordOptions record_options =
     {false, false, false, false, {test_topic_}, {}, {}, {}, {}, {}, {}, {}, "rmw_format", 100ms};
     storage_options_.snapshot_mode = snapshot_mode_;
-    storage_options_.max_cache_size = 200;
+    storage_options_.max_cache_size = 700;
     recorder_ = std::make_shared<rosbag2_transport::Recorder>(
       std::move(writer_), storage_options_, record_options, recorder_name_);
     recorder_->record();

